server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/DLPConfigurationRoutes.java

```diff
@@ -0,0 +1,217 @@
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static org.apache.james.webadmin.Constants.EMPTY_BODY;
+import static org.apache.james.webadmin.Constants.JSON_CONTENT_TYPE;
+import static org.apache.james.webadmin.Constants.SEPARATOR;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.apache.james.core.Domain;
+import org.apache.james.dlp.api.DLPConfigurationItem;
+import org.apache.james.dlp.api.DLPConfigurationStore;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.api.DomainListException;
+import org.apache.james.webadmin.Routes;
+import org.apache.james.webadmin.dto.DLPConfigurationDTO;
+import org.apache.james.webadmin.utils.ErrorResponder;
+import org.apache.james.webadmin.utils.ErrorResponder.ErrorType;
+import org.apache.james.webadmin.utils.JsonExtractor;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.github.steveash.guavate.Guavate;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiImplicitParam;
+import io.swagger.annotations.ApiImplicitParams;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import spark.HaltException;
+import spark.Request;
+import spark.Service;
+
+@Api(tags = "DLPRules")
+@Path(DLPConfigurationRoutes.BASE_PATH)
+@Produces(JSON_CONTENT_TYPE)
+public class DLPConfigurationRoutes implements Routes {
+
+    static final String BASE_PATH = "/dlp/rules";
+
+    private static final String DOMAIN_NAME = ":senderDomain";
+    private static final String SPECIFIC_DLP_RULE_DOMAIN = BASE_PATH + SEPARATOR + DOMAIN_NAME;
+
+    private final JsonTransformer jsonTransformer;
+    private final DLPConfigurationStore dlpConfigurationStore;
+    private final JsonExtractor<DLPConfigurationDTO> jsonExtractor;
+    private final DomainList domainList;
+
+    private Service service;
+
+    @Inject
+    public DLPConfigurationRoutes(DLPConfigurationStore dlpConfigurationStore, DomainList domainList, JsonTransformer jsonTransformer) {
+        this.dlpConfigurationStore = dlpConfigurationStore;
+        this.domainList = domainList;
+        this.jsonTransformer = jsonTransformer;
+        this.jsonExtractor = new JsonExtractor<>(DLPConfigurationDTO.class, new GuavaModule());
+    }
+
+    @Override
+    public void define(Service service) {
+        this.service = service;
+
+        defineStore();
+
+        defineList();
+
+        defineClear();
+    }
+
+    @PUT
+    @Path("/{senderDomain}")
+    @ApiOperation(value = "Store a of dlp config for given senderDomain")
```

Comment:
DLP? maybe a good place to define what it stands for?

Reply:
I think, we can add a description for that at the class level:
`@ApiModel(description="DLP stands for...")`